### PR TITLE
Update SIGPROT signal codes

### PR DESCRIPTION
**Pull request by bsdjhb**

### CODE REVIEW

File: sys/sys/signal.h [hunk 1]
```diff
@@ -437,9 +437,9 @@ struct siginfo64 {
 #define	PROT_CHERI_PERM		5	/* Capability permission fault	*/
 #define	PROT_CHERI_IMPRECISE	7	/* Imprecise bounds fault	*/
 #define	PROT_CHERI_STORELOCAL	8	/* Store-local fault		*/
-#define	PROT_CHERI_CCALL	9	/* CCall fault			*/
-#define	PROT_CHERI_CRETURN	10	/* CReturn fault		*/
+#define	PROT_CHERI_CINVOKE	9	/* CInvoke fault		*/
 #define	PROT_CHERI_SYSREG	11	/* Capability system register fault */
+#define	PROT_CHERI_UNALIGNED_BASE 12	/* Unaligned base address.      */
```
Comment (bsdjhb) [inline sys/sys/signal.h:442]: Adding a new `PROT_CHERI_*` for this is a bit hacky.   None of the others are good fits though.

**Comment by bsdjhb** [comment]
@jrtc27 could you or @PeterRugg confirm that the list of valid CHERI exception codes in the updated sys/riscv/include/cherireg.h is correct?

**Comment by PeterRugg** [comment]
> @jrtc27 could you or @PeterRugg confirm that the list of valid CHERI exception codes in the updated sys/riscv/include/cherireg.h is correct?

Seems to line up with what we have in hardware

File: sys/riscv/include/cherireg.h [hunk 2]
```diff
@@ -193,17 +191,12 @@
 #define	CHERI_EXCCODE_STORE_LOCALCAP	0x16
 #define	CHERI_EXCCODE_PERM_SEAL		0x17
 #define	CHERI_EXCCODE_SYSTEM_REGS	0x18
-#define	CHERI_EXCCODE_PERM_CCALL	0x19
-#define	CHERI_EXCCODE_CCALL_IDC		0x1a
+#define	CHERI_EXCCODE_PERM_CINVOKE	0x19
+#define	CHERI_EXCCODE_CINVOKE_IDC	0x1a
```
Comment (jrtc27) [inline sys/riscv/include/cherireg.h:195]: This should be deprecated and removed. It only existed whilst CCallFast had a delay slot in CHERI-MIPS, raised when the delay slot tried to access IDC, but (a) the delay slot later went away (b) CHERI-MIPS itself went away. We should update cheri-specification accordingly too.

Comment (bsdjhb) [reply sys/riscv/include/cherireg.h:195]: I've added the ISA change to the de-MIPS PR.

**Pull request merged**